File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.js
var processStr = function (s) {
    const result = []
    for (const char of s) {
        if (char >= 'a' && char <= 'z') {
            result.push(char)
        } else if (char === '*') {
            if (result.length) {
                result.pop()
            }
        } else if (char === '#') {
            result.push(...result)
        } else if (char === '%') {
            result.reverse()
        }
    }
    return result.join("")
};

File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        vector<char> result;
        for (char& ch : s) {
            if (islower(ch)) {
                result.push_back(ch);
            } else if (ch == '*') {
                if (!result.empty()) {
                    result.pop_back();
                }
            } else if (ch == '#') {
                vector<char> copy = result;
                result.insert(result.end(), copy.begin(), copy.end());
            } else if (ch == '%') {
                reverse(result.begin(), result.end());
            }
        }
        return string(result.begin(), result.end());
    }
};